.github/workflows/agent-validation.yml

```
@@ -0,0 +1,86 @@
+            exit 1
+          fi
+          
+          # Extract task id from PR title or branch name
+          TASK_ID=$(printf "%s\n%s" "$PR_TITLE" "$HEAD_REF" | sed -n 's/.*\(T-[0-9][0-9][0-9]\).*/\1/p' | head -n1 || true)
+          echo "Detected TASK_ID: ${TASK_ID:-none}"
+          
+          # If task rule exists, enforce globs for changed files
+          if [ -n "${TASK_ID:-}" ] && [ -f ".cursor/rules/tasks/${TASK_ID}.mdc" ]; then
+            echo "Using rule: .cursor/rules/tasks/${TASK_ID}.mdc"
+            python3 - <<'PY'
+import os, fnmatch, sys
```

actionlint (1.7.7)

could not parse as YAML: yaml: line 51: could not find expected ':'

(syntax-check)